src/cmdstan/command_helper.hpp

```diff
@@ -738,14 +738,13 @@ void check_file_config(argument_parser &parser) {
 }
 
 std::vector<std::string> make_filenames(const std::string &filename,
+                                        const std::string &tag,
                                         const std::string &type,
                                         unsigned int num_chains,
                                         unsigned int id) {
   std::vector<std::string> names(num_chains);
   auto base_sfx = get_basename_suffix(filename);
-  if (base_sfx.second.empty()) {
-    base_sfx.second = type;
-  }
+  base_sfx.second = type;
```

Comment:
As noted, this is breaking tests (and would possibly break users code, if they were expecting the old behavior for filenames).

I think you could either add something like an `bool override_type = false` argument to keep the old behavior for all cases besides the pathfinder JSONs, or reconsider if reusing the output filename is the right abstraction